src/Baseline/Reflection/PropertyChain.cs

```diff
@@ -60,16 +57,15 @@ public Type OwnerType
                 var last = _valueGetters.Last();
                 if (last is MethodValueGetter || last is IndexerValueGetter)
                 {
-                    var nextUp = _chain.Reverse().Skip(1).FirstOrDefault() as PropertyValueGetter;
-                    if (nextUp != null)
+                    if (_chain.Reverse().Skip(1).FirstOrDefault() is PropertyValueGetter nextUp)
                     {
                         return nextUp.PropertyInfo.PropertyType;
                     }
                 }
 
-                var propertyGetter = _chain.Last() as PropertyValueGetter;
-                if (propertyGetter != null) return propertyGetter.PropertyInfo.PropertyType;
-                return InnerProperty != null ? InnerProperty.DeclaringType : null;
+                if (_chain.Last() is PropertyValueGetter propertyGetter) return propertyGetter.PropertyInfo.PropertyType;
+                
+                return InnerProperty?.DeclaringType!;
```

Comment:
It appears possible that `InnerProperty` might be null when this property is read, however would it be ever be null in practice? Need some additional domain knowledge to decide if `OwnerType` should be nullable.